### PR TITLE
fix: Handle error in trigger requests Appsmith AI

### PR DESCRIPTION
**Pull request by vivonk**
## Description
Handling unexpected errors in trigger requests in Appsmith AI plugin

#### PR fixes following issue(s)
Fixes #

#### Type of change
- Bug fix (non-breaking change which fixes an issue)

#### How Has This Been Tested?
- [x] Manual

## Automation

/ok-to-test tags="@tag.Datasource"

#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **Bug Fixes**
	- Improved error handling for file uploading and listing to enhance stability and user experience.

> [!IMPORTANT]  
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/8245477758>
> Commit: `8a36d1b0216f86c18ebee541fbad2fec9e66dfb9`
> Cypress dashboard url: <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=8245477758&attempt=1" target="_blank">Click here!</a>
> All cypress tests have passed 🎉🎉🎉

### CODE REVIEW

**Comment by Nikhil-Nandagopal** [comment]
Closing this as it does not have an appropriate label associated with a pod

File: app/server/appsmith-plugins/appsmithAiPlugin/src/main/java/com/external/plugins/AppsmithAiPlugin.java [hunk 1]
```diff
@@ -87,6 +87,16 @@ public Mono<TriggerResultDTO> trigger(
                             TriggerResultDTO triggerResultDTO = new TriggerResultDTO();
                             triggerResultDTO.setTrigger(response);
                             return Mono.just(triggerResultDTO);
+                        })
+                        .onErrorResume(error -> {
+                            log.error(
```
Comment (nidhi-nair) [inline app/server/appsmith-plugins/appsmithAiPlugin/src/main/java/com/external/plugins/AppsmithAiPlugin.java:92]: Should we be pulling this out into a separate method for reusability?

Comment (vivonk) [reply app/server/appsmith-plugins/appsmithAiPlugin/src/main/java/com/external/plugins/AppsmithAiPlugin.java:92]: Sure, making this change.

Comment (nidhi-nair) [reply app/server/appsmith-plugins/appsmithAiPlugin/src/main/java/com/external/plugins/AppsmithAiPlugin.java:92]: A `Mono` error type does not have to be classified as a separate type from the actual type you are dealing with. Think of them as `RuntimeException` thrown from a class. Just like how you never have to specify its possibility in the function signature, a `Mono` error is just a type of signal that is sent from the reactive flow. (on the lines of `complete`, or `next`. Check [this](https://projectreactor.io/docs/core/release/api/reactor/core/publisher/Signal.html) doc for deets.

**Comment by nidhi-nair** [comment]
Removing review request so fresh changes show up in the queue correctly.

**Pull request merged**